cli/tools/jupyter/server.rs

```diff
@@ -410,87 +410,81 @@ impl JupyterServer {
-        let name = if let Some(class_name) = &exception.class_name {
-            format!("Uncaught {}", class_name)
-        } else {
-            "Unknown exception".to_string()
-        };
+      let (name, message, stack) =
+        if let Some(exception) = exception_details.exception {
+          let result = self
+            .repl_session
+            .call_function_on_args(
+              r#"
+          function(object) {
+            if (object instanceof Error) {
+              const name = "name" in object ? String(object.name) : "";
+              const message = "message" in object ? String(object.message) : "";
+              const stack = "stack" in object ? String(object.stack) : "";
+              return JSON.stringify({ name, message, stack });
+            } else {
+              const message = String(object);
+              return JSON.stringify({ name: "", message, stack: "" });
+            }
+          }
+        "#
+              .into(),
+              &[exception],
+            )
+            .await?;
 
-        let remote_object = exception;
-
-        let value: serde_json::Value = match remote_object.kind.as_str() {
-            "string" => remote_object.value.clone().unwrap_or_else(|| json!(" ")),
-            "object" => {
-                match remote_object.subtype.as_deref() {
-                    Some("error") => {
-                        if let Some(message) = remote_object.value.clone().and_then(|v| v.get("message").cloned()) {
-                          json!(message)
-                        } else if let Some(description) = remote_object.description.clone() {
-                          json!(description)
-                        } else {
-                          json!("wat")
-                        }
-                    },
-                    _ => remote_object.value.clone().unwrap_or_else(|| json!(" ")),
-                }
-            },
-            _ => remote_object.value.clone().unwrap_or_else(|| json!(" ")),
+          match result.result.value {
+            Some(serde_json::Value::String(str)) => {
+              if let Ok(object) =
+                serde_json::from_str::<HashMap<String, String>>(&str)
+              {
+                let get = |k| object.get(k).cloned().unwrap_or_default();
+                (get("name"), get("message"), get("stack"))
+              } else {
+                eprintln!("Unexpected result while parsing JSON {str}");
+                ("".into(), "".into(), "".into())
+              }
+            }
+            _ => {
+              eprintln!("Unexpected result while parsing exception {result:?}");
+              ("".into(), "".into(), "".into())
+            }
+          }
+        } else {
+          eprintln!("Unexpectedly missing exception {exception_details:?}");
+          ("".into(), "".into(), "".into())
         };
 
-        (name, value)
+      let stack = if stack.is_empty() {
+        format!(
+          "{}\n    at <unknown>",
```

Comment:
I wonder if we should use like `at <In[3]>` or something like that. Something for later dev experience to hammer out.

Reply:
In this particular case we actually have no information about where the stack trace was thrown, but I suspect that once we get execution indexes working we might be able to replace the `at <anonymous>:2:7` with a much cleaner `at <In[3]>:2:7`.

Example:

```
13:14 $ deno
Deno 1.36.3
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> throw "a party"

Uncaught "a party"
```

vs.

```
> throw new TypeError()
Uncaught TypeError
    at <anonymous>:2:7
```